### ISSUE TITLE
[v6] navigate({ search }) modifies path

### ISSUE THREAD

**Issue by onionhammer** [remix-run/react-router#7496]
Version: 6.0.0-beta.0

For URL: some/path

Code:
```ts
const navigate    = useNavigate();
...
navigate({ search: new URLSearchParams({ "foo": "1", "bar": "2" }).toString() });

```

Expected URL: some/path?foo=1&bar=2

**Actual URL**: some/pathfoo=1&bar=2

Workaround (hack):

```ts
const search = new URLSearchParams({ "foo": "1", "bar": "2" }).toString();
navigate({ search: search.length ? "?" + search : "" });
```

**Comment by StringEpsilon** [comment]
Possibly related: https://github.com/ReactTraining/history/issues/813

**Comment by onionhammer** [comment]
@StringEpsilon yep, that looks like its related, assuming navigate() uses that function internally

**Comment by onionhammer** [comment]
Afaik this has not been fixed

**Comment by timdorr** [comment]
Closed by #7953
